<launch>
  <!-- launch the multi map server for the world -->
  <arg name="multimap_file" default="$(find utexas_gdc)/maps/real/multimap/multimap.yaml" />
  <node name="multi_level_map_server" pkg="multi_level_map_server" type="multi_level_map_server">
    <param name="map_file" value="$(arg multimap_file)" />
  </node>

  <!-- Set the default level we'd like this robot to start on -->
  <param name="level_mux/default_current_level" value="3rdFloor" />

  <include file="$(find bwi_object_search)/launch/includes/segbot.launch.xml">
    <arg name="laser_max_range" value="20.0" />
  </include>

  <include file="$(find segbot_bringup)/launch/segbot_v2.launch" />

  <include file="$(find bwi_logging)/launch/record.launch" />

  <!-- <node pkg="rviz" type="rviz" name="rviz" -->
  <!--   args="-d $(find ar_pose)/launch/live_single.rviz"/> -->
  <!-- <node pkg="tf" type="static_transform_publisher" name="world_to_cam" -->
  <!--   args="0 0 0.5 -1.57 0 -1.57 world camera 10" /> -->

  <!-- Not sure why I cannot change the camera using this, but works with logitech webcam -->
  <!-- <node ns="usb_camera" pkg="image_proc" type="image_proc" name="image_proc"/> -->
  <!-- <node ns="usb_camera" pkg="uvc_camera" type="uvc_camera_node" name="usb_camera" -->
  <!--   output="screen"> -->
  <!--   <param name="width" type="int" value="640" /> -->
  <!--   <param name="height" type="int" value="480" /> -->
  <!--   <param name="fps" type="int" value="30" /> -->
  <!--   <param name="frame_id" type="string" value="usb_camera" /> -->
  <!--   <param name="device" type="string" value="/dev/video1" /> -->
  <!--   <param name="camera_info_url" type="string" -->
  <!--     value="file:///home/users/wxie/catkin_ws/src/usb_cam/calibration.yaml" /> -->
      <!-- value="file://$(find usb_cam)/calibration.yaml" /> -->
  <!-- </node> -->

  <node name="usb_camera" pkg="usb_cam" type="usb_cam_node" output="screen" >
    <param name="camera_name" value="usb_camera" />
    <param name="video_device" value="/dev/video0" />
    <param name="image_width" value="640" />
    <param name="image_height" value="480" />
    <param name="pixel_format" value="yuyv" />
    <param name="camera_frame_id" value="usb_camera_link" />
    <param name="io_method" value="mmap"/>
    <param name="autofocus" value="false" />
    <param name="camera_info_url" value="file://$(find bwi_object_search)/calibration/bender_laptop_camera.yaml"/>
  </node>
  <node ns="usb_camera" pkg="image_proc" type="image_proc" name="image_proc"/>

  <!-- Single -->
  <!-- <node name="ar_pose" pkg="ar_pose" type="ar_single" respawn="false" -->
  <!--   output="screen"> -->
  <!--   <param name="marker_pattern" type="string" -->
  <!--     value="$(find ar_pose)/data/4x4/4x4_83.patt"/> -->
  <!--   <param name="marker_width" type="double" value="138.4"/> -->
  <!--   <param name="marker_center_x" type="double" value="0.0"/> -->
  <!--   <param name="marker_center_y" type="double" value="0.0"/> -->
  <!--   <param name="threshold" type="int" value="80"/> -->
  <!--   <param name="use_history" type="bool" value="true"/> -->
  <!-- </node> -->

  <!-- Multi -->
  <node name="ar_pose" pkg="ar_pose" type="ar_multi" respawn="false"
    output="screen">
    <param name="marker_pattern_list" type="string"
      value="$(find bwi_object_search)/data/object_list.txt"/>
    <param name="threshold" type="int" value="100"/>
  </node>

</launch>
